File: app/src/main/java/com/cangwang/modulebus/ViewPagerAdapter.java
package com.cangwang.modulebus;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import java.util.List;

/**
 * Created by air on 16/7/6.
 */
public class ViewPagerAdapter extends FragmentPagerAdapter {
    private List<String> titles;
    private List<Fragment> fragmentList;

    public ViewPagerAdapter(FragmentManager fm){
        super(fm);
    }

    public ViewPagerAdapter(FragmentManager fm, List<Fragment> lf, List<String> titles){
        super(fm);
        this.titles = titles;
        fragmentList = lf;
    }

    @Override
    public Fragment getItem(int position) {
        return fragmentList.get(position);
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return titles.get(position);
    }

    @Override
    public int getCount() {
        return fragmentList.size();
    }
}

File: app/src/main/java/com/cangwang/modulebus/ViewPagerAdapter.kt
package com.cangwang.modulebus

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

/**
 * Created by air on 16/7/6.
 */
class ViewPagerAdapter : FragmentPagerAdapter {
    private var titles: List<String>? = null
    private var fragmentList: List<Fragment>? = null

    constructor(fm: FragmentManager?) : super(fm!!) {}
    constructor(fm: FragmentManager?, lf: List<Fragment>?, titles: List<String>?) : super(fm!!) {
        this.titles = titles
        fragmentList = lf
    }

    override fun getItem(position: Int): Fragment {
        return fragmentList!![position]
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return titles!![position]
    }

    override fun getCount(): Int {
        return fragmentList!!.size
    }
}